public/app/features/alerting/unified/hooks/useCombinedRule.ts

```diff
@@ -67,7 +67,7 @@ export function useCloudCombinedRulesMatching(
   const { dsFeatures, isLoadingDsFeatures } = useDataSourceFeatures(ruleSourceName);
 
   const {
-    currentData: promRuleNs = [],
```

Comment:
The infinite re-render was caused by this unstable array reference which is added to the dependencies for the `useAsync` hook, which ends up causing an infinite re-render.